src/main/markdown/eco/eco-system.md

```diff
@@ -0,0 +1,263 @@
+# GWT Eco System
+
+Today, the GWT eco system is not only that, what comes with the framework. Usually, projects use a lot of third party
+dependencies to improve there development experience, add features or prefer a different approach to do things. 
+
+But, before we start a little history of GWT, the current state and a foresight:
+
+## History
+
+First and foremost, GWT is a Java to JavaScript transpiler. The basic idea is to use a type-safe and object oriented
+language to create applications which can ran natively in the browser. During development, developers benefit from the 
+existing Java tool chain, powerful IDEs and awesome refactoring abilities. In production the transpiler creates highly
+optimized and obfuscated JavaScript.
+
+In 2006, as GWT emerges, things were different. JavaScript depends on the browser and Java were in an
+early state of evolution. That's one of the reasons why we have permutations and generators in GWT. Also there no
+third-party-dependencies. Everything had to be provided by the framework. With GWT 2,  a lot of new modules like: editor, uibinder, etc. 
+were added to the SDK.
+
+Today, things are different. Java has evolved. New language feature, annotation processors, etc. are state of the art
+and the gaps of the JavaScript engines in the different browser are no longer a pain point. Third party dependencies add a lot of 
+things that GWT does not cover or offer an alternative implementation of existing implementations.
+
+Also, Google transfers the lead of the GWT project to an Open Source community.
+
+Looking at the next generation transpiler from Google (J2CL) there are no longer permutations, generators or modules 
+available. Today, it is only a transpiler. In preparation of using this new transpiler and keeping in mind the huge
+amount of legacy applications the community starts to migrate the GWT modules in 2019. These migrated modules no longer 
+use generators (instead they are using annotation processors) and JSNI (JavaScript Native Interface). Now, they are
+using JsInterop. On the long term, using this new modules, will enable applications to switch to the new transpiler
+very smoothly.
```

Comment:
I think this paragraph needs more justification for GWT 2 projects to use the modules.

Specifically:
 * Java's principles include keeping projects maintainable over long periods of time: While changes like the addition of JPMS have required some projects to make changes, those are infrequent and usually not too major.
 * Likewise, GWT tries to avoid breaking changes, enabling projects to continue to compile old projects using old APIs to still work with new browsers and new Java releases. This cannot always be perfect (code that assumes old browsers obviously will not work), but it will support teams who can update to get security fixes, performance improvements, and new browser support with minimal effort of their own.
 * However, this limitation on breaking existing code makes the repository "append-only" - we cannot remove old features easily. For the compiler itself, this is largely not a problem, but for gwt-user, any change might break existing applications. It is more important to support old projects updating in a low- or no-effort manner, than to continue to mutate the APIs of these projects.
 * Additionally, there is an unclear dependency between gwt-user.jar and gwt-dev.jar - if a new compiler offers compiled size improvements, it is usually not possible to update only gwt-dev.jar and leave and old, working version of gwt-user.jar. (in contrast, it is usually possible to keep an old version of gwt-servlet.jar, not that we encourage this).
 * Instead, moving the "user" code (aka GWT modules) out to separate modules enables teams to keep their runtime code at an older version not tied to the compiler. Maintainers of the modules can likewise experiment with APIs or remove dead code, and regardless of compiler version or implementation, downstream projects can adapt to those changes.